pkg/router/controller/factory/factory.go

```diff
@@ -25,30 +25,32 @@ import (
 // controller. It supports optional scoping on Namespace, Labels, and Fields of routes.
 // If Namespace is empty, it means "all namespaces".
 type RouterControllerFactory struct {
-	KClient        kclient.EndpointsNamespacer
-	OSClient       osclient.RoutesNamespacer
-	Namespaces     controller.NamespaceLister
-	ResyncInterval time.Duration
-	Namespace      string
-	Labels         labels.Selector
-	Fields         fields.Selector
+	EndpointsNamespacer kclient.EndpointsNamespacer
```

Comment:
> Is there a reason we aren't writing the annotation to the endpoints object instead of the service?

We do write the annotation to the endpoints object...

> would that be sufficient to know here?

No, you have to know the service IP so that the router can target requests to the unidling proxy (instead of just returning the "no endpoints" HTTP error) for idled services.